lib/backend/model/resource.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/namespace"
+)
+
+var (
+	matchGlobalResource     = regexp.MustCompile("^/calico/resources/v2/([^/]+)/([^/]+)$")
+	matchNamespacedResource = regexp.MustCompile("^/calico/resources/v2/([^/]+)/([^/]+)/([^/]+)$")
+	kindToType              = map[string]reflect.Type{
+		strings.ToLower(apiv2.KindBGPPeer):          reflect.TypeOf(apiv2.BGPPeer{}),
+		strings.ToLower(apiv2.KindIPPool):           reflect.TypeOf(apiv2.IPPool{}),
+		strings.ToLower(apiv2.KindNode):             reflect.TypeOf(apiv2.Node{}),
+		strings.ToLower(apiv2.KindHostEndpoint):     reflect.TypeOf(apiv2.HostEndpoint{}),
+		strings.ToLower(apiv2.KindNetworkPolicy):    reflect.TypeOf(apiv2.NetworkPolicy{}),
+		strings.ToLower(apiv2.KindNetworkPolicy):    reflect.TypeOf(apiv2.NetworkPolicy{}),
```

Comment:
shouldnt this be `kindGlobalNetworkPolicy`?

Reply:
Yup!